### PR TITLE
feat(component): implement ComponentName parser and validator.

### LINKED ISSUE #4442: feat: Component Model: implement the ComponentName
### Summary

Implement the parser and validator for `exportname` and `importname`

Major work : https://github.com/WasmEdge/WasmEdge/pull/4430

### Details

Some details in Import and Export Definitions we need to implement.

There are some subtasks :
* interfaceversion (https://semver.org/)
* hashname

It would be a pleasure if you’d like to make a contribution.

### Appendix

https://github.com/WebAssembly/component-model/blob/main/design/mvp/Explainer.md#import-and-export-definitions

https://semver.org/

### PR DESCRIPTION
**Pull request by Harsh-6291**
## what does this pr do
This PR implements the ComponentName parser and validator for the WebAssembly Component Model in WasmEdge. fixing #4442

## Key Changes
* AST: Added ComponentName struct supporting plain names, scoped names (namespace:package@version), and hash names (integrity-sha256:...).

* Loader: Implemented loadExternName parsing logic with integrity prefix detection.

* Validator: Added validateComponentName with semver (2+ dots, [0-9a-zA-Z.-+]) and hash validation.

* Executor: Updated instantiation/lookup logic for new name structure.

## tests
Successfully validated the implementation against 15 test cases using wasm-tools generated components:
 * Test 1-5: Initial validation suite (Plain, Valid/Invalid Scoped, Valid/Invalid Hash).
 * Test 6-15: Edge case suite (Multiple @, 4+ part semver, pre-release tags, ambiguous integrity- prefixes, etc.).
 
### Fixes #4442

### CODE REVIEW

File: lib/executor/instantiate/component/component_instance.cpp [hunk 1]
```diff
@@ -165,30 +165,31 @@ Executor::instantiate(Runtime::Instance::ComponentInstance &CompInst,
           switch (Sort.getCoreSortType()) {
           case AST::Component::Sort::CoreSortType::Func:
             Comp->addCoreFunction(CompInst.getCoreFunction(Idx));
-            Comp->exportCoreFunction(Exp.getName(), CoreExpIdx[0]);
+            Comp->exportCoreFunction(Exp.getName().Name, CoreExpIdx[0]);
```
Comment (dannypsnl) [inline lib/executor/instantiate/component/component_instance.cpp:168]: If we have to use `getName()` like this, shouldn't we just change the signature of these functions?

File: lib/executor/instantiate/component/component_import.cpp [hunk 2]
```diff
@@ -62,15 +63,17 @@ Executor::instantiate(Runtime::Instance::ComponentImportManager &ImportMgr,
...
     case AST::Component::ExternDesc::DescType::InstanceType: {
       // TODO: COMPONENT - type matching for the instance type.
       auto CompName = Import.getName();
-      const auto *ImportedCompInst = ImportMgr.findComponentInstance(CompName);
+      const auto *ImportedCompInst =
+          ImportMgr.findComponentInstance(CompName.Name);
       if (unlikely(ImportedCompInst == nullptr)) {
         spdlog::error(ErrCode::Value::UnknownImport);
-        spdlog::error("    component name: {}"sv, CompName);
+        spdlog::error("    component name: {}"sv, CompName.Name);
```
Comment (dannypsnl) [inline lib/executor/instantiate/component/component_import.cpp:76]: You can attach pretty print implementation to a class so you don't need to access its field for printing.

**Comment by Harsh-6291** [comment]
hey @dannypsnl  i have made those changes now

**Comment by LFsWang** [comment]
@Harsh-6291 I think this should be integrated with the current ```ComponentName ``` class. However, moving the current class to the parsing state would also be fine.

https://github.com/WasmEdge/WasmEdge/blob/master/lib/validator/component_name.cpp

**Comment by hydai** [comment]
I am closing this because the basic checks have failed, and some of the review comments remain unresolved. Additionally, there appears to be inactivity. Thank you for your contribution. Please feel free to reopen it once your PR is ready for review again.

**Comment by Harsh-6291** [comment]
Hi @hydai, 
I've addressed all the feedback and fixed the CI failures. The implementation is now ready for review.

**Comment by Harsh-6291** [comment]
hey @hydai  can you please reopen this pr
